src/crimson/os/seastore/transaction_manager.h

```diff
@@ -433,8 +433,7 @@ class TransactionManager : public ExtentCallbackInterface {
     ceph_assert(total_remap_len < original_len);
 #endif
 
-    // FIXME: paddr can be absolute and pending
-    ceph_assert(pin->get_val().is_absolute());
+    ceph_assert(pin->is_stable());
```

Comment:
I think we need to determine whether an unstable pin can be remapped or not. The test_remap_pin test case involves multiple remaps within a single transaction. This leads to a failure in the second remap attempt.

Reply:
Thanks for the investigation!

For the real (non-test) use case, overwriting a pending extent shouldn't involve remapping -- it is better for data to be overwritten in the pending buffer directly. IIUC this is why this ceph-assert was added initially. Also refer to the decision made in `overwrite_plan_t`.

So, probably we can just drop this check (and merge all into 1 commit) as testing remapping pending extents doesn't look unreasonable to me.